lib/vbridge/VBridge.ts

```diff
@@ -0,0 +1,580 @@
+  //   for (const key of bridgeConfig.anchors.keys()) {
+  //     const createdAnchorIdentifier = Bridge.createAnchorIdentifier(key);
+  //     if (createdAnchorIdentifier) {
+  //       identifiedAnchors.push(createdAnchorIdentifier);
+  //     }
+  //   }
+
+  //   // loop through and group anchors by their identifiers
+  //   let groupLinkedAnchors: Anchor[][] = [];
+
+  //   for (const anchor of identifiedAnchors) {
+  //     let anchorGroup: Anchor[] = [];
+
+  //     for (const linkableAnchor of identifiedAnchors) {
+  //       if (
+  //         anchor.tokenName == linkableAnchor.tokenName && 
+  //         anchor.anchorSize == linkableAnchor.anchorSize
+  //       ) {
+  //         anchorGroup.push(bridgeConfig.anchors.get(Bridge.createAnchorIdString(linkableAnchor))!);
+  //       }
+  //     }
+
+  //     groupLinkedAnchors.push(anchorGroup);
+  //   }
+
+  //   const linkedAnchors = await Bridge.createLinkedAnchorMap(groupLinkedAnchors);
+    
+  //   return new Bridge(bridgeConfig.bridgeSides, bridgeConfig.webbTokenAddresses, linkedAnchors, bridgeConfig.anchors);
+  // }
+
+  public static async deployVBridge(vBridgeInput: VBridgeInput, deployers: DeployerConfig): Promise<VBridge> {
+    
+    let webbTokenAddresses: Map<number, string> = new Map();
+    let vBridgeSides: Map<number, VBridgeSide> = new Map();
+    let vAnchors: Map<string, VAnchor> = new Map();
+    // createdAnchors have the form of [[Anchors created on chainID], [...]]
+    // and anchors in the subArrays of thhe same index should be linked together
+    let createdVAnchors: VAnchor[][] = [];
+
+    for (let chainID of vBridgeInput.chainIDs) {
+      const adminAddress = await deployers[chainID].getAddress();
+
+      // Create the bridgeSide
+      const vBridgeInstance = await VBridgeSide.createVBridgeSide(
+        [adminAddress],
+        1,
+        0,
+        100,
+        deployers[chainID],
+      );
+
+      vBridgeSides.set(chainID, vBridgeInstance);
+      console.log(`vBridgeSide address on ${chainID}: ${vBridgeInstance.contract.address}`);
+
+      // Create the Hasher and Verifier for the chain
+      const hasherFactory = await getHasherFactory(deployers[chainID]);
+      let hasherInstance = await hasherFactory.deploy({ gasLimit: '0x5B8D80' });
+      await hasherInstance.deployed();
+
+      const verifier = await Verifier.createVerifier(deployers[chainID]);
+      let verifierInstance = verifier.contract;
+
+      // Check the addresses of the asset. If it is zero, deploy a native token wrapper
+      let allowedNative: boolean = false;
+      for (const tokenToBeWrapped of vBridgeInput.vAnchorInputs.asset[chainID]!) {
+        // If passed '0' or zero address, token to be wrapped should support native.
+        if (checkNativeAddress(tokenToBeWrapped)) {
+          allowedNative = true;
+        }
+      }
+
+      let tokenInstance: GovernedTokenWrapper = await GovernedTokenWrapper.createGovernedTokenWrapper(
```

Comment:
We should support deploying a bridge with an existing token already too. For example, we already have a bridge for webbETH-test-1 live on ETH testnets and can use this same token for the variable bridge now, as long as we give the proper minting permissions to the VAnchor